### DISCUSSION TITLE
how to add Outlook, Teams and Yammer application data into the PnP Search results

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2295] [Q&A]
Hi, I have tried to add Outlook email conversations/ content into PnP search results web part with adding a entity type = message in search results webpart but always I'm ending with error and couldn't able to fetch the details.

Could you please help us more on what is entity type = "message" and "event" and how to fetch the search records.

thank you

**Comment by wobba** [comment]
You can surface entities from the Graph API. You have to approve of the Graph scopes to start with as mentioned at https://microsoft-search.github.io/pnp-modern-search/installation, then you can start configuring the web part.

We should probably have a tutorial on this setup, but need someone to contribute them.

**Reply by GangadharMFord** [reply]
thank you @wobba for your replay and this will greatly help to reduce to use Microsoft search at Bing.
Any update on this ask please ?

**Reply by wobba** [reply]
When someone have time docs will be updated. What is blocking you? Graph documentation can be found at https://learn.microsoft.com/en-us/graph/search-concept-messages
